gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -37,31 +44,58 @@ public class GoogleCloudStorageGrpcIntegrationTest {
 
   private static final String BUCKET_NAME = BUCKET_HELPER.getUniqueBucketName("shared");
 
-  private static GoogleCloudStorage createGoogleCloudStorage() throws IOException {
+  private final boolean tdEnabled;
+
+  @Parameters
+  // We want to test this entire class with both gRPC-LB and Traffic Director
+  public static Iterable<Boolean> tdEnabled() {
+    return Arrays.asList(false, true);
+  }
+
```

Comment:
This change covers testing with gslb and td with fallback. -- Since the env is not set as part of presubmit.

Effectively in one environment, we are testing TD with or without fallback, but never both. Can we cover all scenarios ?

Reply:
Setting TD should use direct path (not fallback) wherever it can, so it is not always with fallback. So if you run from a VM directpath should be used, it is just we will not be sure if fallback is used. That's when force-dp come into picture, force-dp was not accessible from where the presubmits are run. VPC-SC tests would have force-dp so we would know there. I am not sure how to make presubmits run with force-dp, and that is out of the scope of this PR because this PR makes sure td enabled/disabled runs for customer or not.